source/ControllableHero.hx

```diff
@@ -0,0 +1,48 @@
+package;
+
+import flixel.FlxSprite;
+import flixel.util.FlxColor;
+import flixel.FlxG;
+
+class ControllableHero extends FlxSprite
+{
+    public function new()
+        {
+            super();
+            makeGraphic(50, 45, FlxColor.WHITE, false);
+        }
+
+
+        override public function update(elapsed:Float) 
+        {
+            if (FlxG.keys.justPressed.UP)
+                {this.y += -1;}
+            if (FlxG.keys.pressed.UP)
+                {this.y += -1;}
+            if (FlxG.keys.justReleased.UP)
+                {this.y += -1;}
```

Comment:
`FlxG.keys.justPressed.<whatever>` will be `true` for just the first `update()` in which that key was detected as "pressed". To note, that means that on the `update` where `justPressed` is `true`, `pressed` will also be `true`. This means we move double on the frame that a button is pressed, which probably isn't the logic that we wanted here.

Similarly, `justReleased` will be `true` after the button is no longer held, meaning that we'll continue to move on the `update` after we stop pressing the direction.

the `justPressed` and `justReleased` inputs are best used for 'one shot' behaviors that you want to trigger at the start or end of the input being pressed. Like, maybe you want to turn the player red when they start moving, then revert them back to a plain color when they stop. You could do it by setting their color every frame while `pressed` is true, but it can be made much cleaner by using the information provided by `justPressed` and `justReleased`. 

In this case, I think we only really want to do this movement when the button is held, so we only need the `pressed` if statements here, the others should be removed (unless there's a special reason to keep them).